opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -1065,14 +1065,15 @@ namespace Opm
         if ( bhp_limit_not_defaulted || !this->wellHasTHPConstraints(summaryState) ) {
             // if the BHP limit is not defaulted or the well does not have a THP limit
             // we need to check the BHP limit
-
+            bool operable_under_bhp = false;
             for (int p = 0; p < this->number_of_phases_; ++p) {
-                const double ipr_rate = this->ipr_a_[p] - this->ipr_b_[p] * bhp_limit;
-                if ( (this->isProducer() && ipr_rate < 0.) || (this->isInjector() && ipr_rate > 0.) ) {
-                    this->operability_status_.operable_under_only_bhp_limit = false;
-                    break;
+                double ipr_rate = this->ipr_a_[p] - this->ipr_b_[p] * bhp_limit;
```

Comment:
The sign of the ipr_rate is unfortunately following the opposite convention as our usual (negative flow is production). We should probably fix this (but not in this PR).